File: 4 - Cuarto Semestre/4 - Estructuras de Datos y Algoritmos/4 - Evaluaciones/E1/billetera.java
public class Billetera {
    private String titular; // Dueño de la cuenta
    private String cuenta; // #Cuenta
    private int saldo; // Dinero

    public Billetera(String titular, String cuenta) {
        this.titular = titular;
        this.cuenta = cuenta;
        this.saldo = 0;
    }

    public void depositar(int monto) { // Método para agregar dinero a la cuenta
        saldo += monto;
    }

    public void pagar(Billetera dest, int monto) { // Método para transferencias
        saldo -= monto;
        dest.depositar(monto);
    }

    @Override
    public boolean equals(Object obj) {
        if (obj == null || getClass() != obj.getClass()) {
            return false;
        }
        Billetera other = (Billetera) obj;
        return saldo == other.saldo;
    } 

    @Override
    public String toString() {
        return "Billetera: titular=" + titular + ", cuenta=" + cuenta + ", saldo=" + saldo;
    }

    public static void main(String[] args) {
        Billetera b1 = new Billetera("Juanito", "123-123");
        Billetera b2 = new Billetera("Anita", "453-234");
        assert b1.equals(b2); 
        assert b1.saldo == 0;
       
        b1.depositar(1000);
        //assert b1.saldo == 1000;
        b1.pagar(b2, 700);
       
        //assert b1.saldo == 300;
        //assert b2.saldo == 700;
        //assert !b1.equals(b2);

        System.out.println("-------------------------------------------");
        System.out.println(b1);
        System.out.println("-------------------------------------------");
        System.out.println(b2);
        System.out.println("-------------------------------------------");
    }
}

File: 4 - Cuarto Semestre/4 - Estructuras de Datos y Algoritmos/4 - Evaluaciones/E1/billetera.py
class Billetera:

    def __init__(self, titular, cuenta):
        self._titular = titular
        self._cuenta = cuenta
        self._saldo = 0

    def depositar(self, monto):
        self._saldo += monto

    def pagar(self, dest, monto):
        self._saldo -= monto
        dest.depositar(monto)

    def __eq__(self, other):
        if other is None:
            return False
        if isinstance(other, Billetera):
            return self._saldo==other._saldo
        return False

    def __str__(self):
        return f"Billetera: titular={self._titular}, cuenta={self._cuenta}, saldo={self._saldo}"

if __name__ == "__main__":
    b1 = Billetera("Juanito", "123-123")
    b2 = Billetera("Anita", "453-234")
    assert(b1==b2)
    assert(b1._saldo == 0)
    b1.depositar(1000)
    assert(b1._saldo == 1000)
    b1.pagar(b2, 700)
    assert(b1._saldo == 300)
    assert(b2._saldo == 700)
    assert(b1!=b2)

    print(b1)
    print(b2)